scimath/Mathematics/FitToHalfStatistics.tcc

```diff
@@ -128,6 +128,7 @@ void FitToHalfStatistics<CASA_STATP>::_getRealMinMax(
         CountedPtr<AccumType>& realMin, CountedPtr<AccumType>& realMax,
     CountedPtr<AccumType> knownMin, CountedPtr<AccumType> knownMax
 ) {
+    _setRange();
```

Comment:
I don't think so. getMinMax() calls ConstrainedRangeStatistics::getMinMax() as necessary. That method calls _setRange() as defined in the appropriate object.  getQuantiles() calls getMinMax() as necessary, which as discussed ultimately calls _setRange() on the appropriate object. It also calls getNPts() as necessary, which will call the ConstrainedRangeStatistics::getNPtps() which calls _setRange(). It then always calls _getRealMinMax(), which is now guaranteed to called _setRange().

The basic idea is that when the corresponding method in ConstrainedRangeStatistics is called, _setRange() will be called from there. The problem was that there were paths in getQuantiles() in which _getRealMinMax() was being called prior to ConstrainedRangeStatistics::getQuantiles() (which calls _setRange()) being called, so that _centerValue had not yet been set.

It is a bit convoluted I admit, but the goal was to minimize unnecessarily repeating code in derived classes and minimizing unnecessary repeated calls. In general, ConstrainedRangeStatistics methods deal with calling _setRange(), but in this case there were paths in which _setRange() needed to be called prior to a call to the corresponding method in the base class.